libr/search/search.c

```diff
@@ -125,250 +132,203 @@ R_API int r_search_hit_new(RSearch *s, RSearchKeyword *kw, ut64 addr) {
-				if (kw->bin_binmask[i] != 0xff) {
-					hasmask = 0;
-					break;
+		i = !s->overlap && from - kw->last < left->len ? kw->last + left->len - from : 0;
+		for (; i + kw->keyword_length <= len1 && i < left->len; i++) {
+			if (brute_force_match(s, kw, left->data, i) != s->inverse) {
+				if (!r_search_hit_new (s, kw, from + i - left->len)) return -1;
+				kw->count++;
+				count++;
+				if (++s->nhits >= s->maxhits) return count;
+				if (!s->overlap) {
+					i += kw->keyword_length - 1;
 				}
 			}
-			if (!hasmask)
-				kw->binmask_length = 0;
 		}
-	}
-#endif
-	for (i = 0; i < len; i++) {
-		r_list_foreach (s->kws, iter, kw) {
-			if (s->inverse && s->nhits > 0) {
-				//eprintf ("nhits = %d\n", s->nhits);
-				return -1;
-			}
-			for (j = 0; j <= kw->distance; j++) {
-				// TODO: refactor: hit = checkKeyword()
-				// TODO: assert len(kw) == len(bm)
-				ut8 ch, ch2;
-				bool match = false;
-				hit = false;
-
-				if (kw->binmask_length > 0) {
-					match = (maskcmp (kw, buf + i, j) > 0);
-				} else {
-					ch = kw->bin_keyword[kw->idx[j]];
-					ch2 = buf[i];
-					if (kw->icase) {
-						ch = tolower (ch);
-						ch2 = tolower (ch2);
-					}
-					match = (ch == ch2);
-				}
-				if (match) {
-					hit = true;
-				} else {
-					if (s->inverse) {
-						if (!r_search_hit_new (s, kw, (ut64)from + i - kw->keyword_length + 1))
-							return -1;
-						kw->idx[j] = 0;
-						//kw->idx[0] = 0;
-						kw->distance = 0;
-						//eprintf ("HIT FOUND !!! %x %x 0x%llx %d\n", ch, ch2, from+i, i);
-						kw->count++;
-						s->nhits++;
-						return 1; // only return 1 keyword if inverse mode
-					}
-					if (kw->distance < s->distance) {
-						kw->idx[kw->distance + 1] = kw->idx[kw->distance];
-						kw->distance++;
-						hit = true;
-					} else {
-						// Works for 0a0a0a0b
-						bool isTail = false;
-						int k = kw->idx[j];
-						if (k > 0) {
-							int q = 1;
-							int model = kw->bin_keyword[0];
-							for (isTail = true; q < k; q++) {
-								if (kw->bin_keyword[q] != model)
-									isTail = false;
-							}
-						}
-						if (isTail) {
-							kw->idx[j]--;
-						} else {
-							kw->idx[j] = 0;
-						}
-						kw->distance = 0;
-						if (kw->binmask_length > 0) {
-							hit = (maskcmp (kw, buf + i, j) > 0);
-						} else {
-							ch = kw->bin_keyword[kw->idx[j]];
-							hit = (ch == ch2);
-						}
-					}
-				}
-				if (hit) {
-					kw->idx[j]++;
-					if (kw->idx[j] == kw->keyword_length) {
-						if (s->inverse) {
-							kw->idx[j] = 0;
-							continue;
-						}
-						if (!r_search_hit_new (s, kw, (ut64)from + i - kw->keyword_length + 1)) {
-							return -1;
-						}
-						kw->idx[j] = 0;
-						kw->distance = 0;
-						kw->count++;
-						count++;
-						if (s->overlap) {
-							if (kw->keyword_length > 1) {
-								i -= kw->keyword_length - 1;
-							}
-						}
-						//s->nhits++;
-					}
+		for (i = 0; i + kw->keyword_length <= len; i++) {
+			if (brute_force_match(s, kw, buf, i) != s->inverse) {
```

Comment:
Fixed to be consistent with other code, though I always think space between function name and `(args` is bizarre...